examples/logo_interpreter.rs

```diff
@@ -0,0 +1,86 @@
+//! To run a LOGO program, run:
+//!     cargo run --example logo_interpreter -- .\examples\my_logo_program.txt
+
+extern crate turtle;
+
+use turtle::Turtle;
+use std::env;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+
+fn main() {
+    let mut turtle = Turtle::new();
+    turtle.set_speed(4);
+
+    let args: Vec<String> = env::args().collect();
+    
+    if args.len() == 2 {
+        let source= parse_source(&parse_file(&args[1]));
+        parse_string(&source, &mut turtle);
+    }
+    else {
+        println!("Pass in a LOGO command txt file.");
+    }
+}
+
+fn parse_string(source: &str, turtle: &mut Turtle) {
+    let ops: Vec<&str> = source.split(' ').collect();
+
+    let mut ins_pointer = 0;
+
+    while let Some(op) = ops.get(ins_pointer) {
```

Comment:
This code seems to be a bit too complicated for parsing logo commands. There is a lot of repetition and I would prefer if we could reduce the amount of code needed overall. What do you think of replacing this with something like the parsing code that I provided you?

Specifically, in my example, I did two things:

1. I split the string into a `VecDeque<&str>` in advance. That collection type allows you to pop values very cheaply and makes it so that you don't have to keep track of the current index yourself.
```rust
let mut cmd_args = buffer.split_whitespace().collect();
handle_command(&mut turtle, &mut cmd_args);
```
2. I then parse each command by looking at its first part and then recursively fetching any remaining parts
```rust
fn handle_command(turtle: &mut Turtle, args: &mut VecDeque<&str>) {
    if args.is_empty() {
        return;
    }
    // We already checked if args is empty, so we can unwrap here without fear
    match args.pop_front().unwrap() {
        "fd" | "forward" => {
            let distance = parse_distance(args.pop_front()
                .expect("Expected a distance value after fd/forward command"));
            turtle.forward(distance)
        },
        _ => unimplemented!(), //TODO: a couple more commands (no need to do everything)
    }
    // Parse any remaining commands on this line
    handle_command(turtle, args);
}

fn parse_distance(s: &str) -> f64 {
    unimplemented!(); //TODO: Parse a distance and panic if it isn't valid
}
```